Hyper/Controllers/Factories/SphericalControllerFactory.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Chunks.ChunkManagement;
+using Chunks.ChunkManagement.ChunkWorkers;
+using Chunks.MarchingCubes;
+using Chunks.MarchingCubes.MeshGenerators;
+using Common;
+using Common.UserInput;
+using Hud.Shaders;
+using Hyper.Shaders.LightSourceShader;
+using Hyper.Shaders.ModelShader;
+using Hyper.Shaders.ObjectShader;
+using Hyper.Transporters;
+using OpenTK.Mathematics;
+
+namespace Hyper.Controllers.Factories;
+internal class SphericalControllerFactory : IControllerFactory
+{
+    private readonly Vector3i[] _sphereCenters;
+
+    private readonly Vector3 _lowerSphereCenter;
+
+    private readonly float _globalScale;
+
+    private readonly Scene _scene;
+
+    private readonly Context _context;
+
+    private readonly IWindowHelper _windowHelper;
+
+    private readonly ScalarFieldGenerator _scalarFieldGenerator;
+
+    public SphericalControllerFactory(Scene scene, Context context, IWindowHelper windowHelper, ScalarFieldGenerator scalarFieldGenerator, float globalScale = 0.05f)
```

Comment:
DO we need constructor. Can we just have static CreateControllers(...)

Reply:
I wanted to have both factories implement a common interface